[IMPROVED] Audited README files for consistency, formatting, and style (#237)
1Password/scim-examples: PREPARATION.md

# Prepare to deploy 1Password SCIM Bridge

*Learn how to prepare your environment and 1Password account to use 1Password SCIM Bridge for automated user provisioning.*

## Overview

1Password SCIM Bridge uses [System for Cross-domain Identity Management](https://en.wikipedia.org/wiki/System_for_Cross-domain_Identity_Management) (SCIM) to act as an intermediary between your identity provider, such as Okta or Azure Directory, and your 1Password instance. It allows you to centralize user management to your identity provider so you can automatically provision and manage users and groups in 1Password based on assignments in your identity provider.

> **Note**
> 

> > In code and the command line, "1Password" is often referred to as `op`. For example, `op-scim`.

Author: ```suggestion
> In code and on the command line, "1Password" is often referred to as `op`. For example, `op-scim`.
```

### Technical components

For general deployment, the SCIM bridge requires these components:

* the `op-scim` service
* a [redis](https://redis.io/) cache
* a domain name (for example, `op-scim-bridge.example.com`)

### DNS record

> You'll need to be able to create a DNS record with the SCIM bridge domain name you want to use. However, you'll need to have the IP address of the host, so the bridge will need to be deployed first, unless you have a static IP already assigned. Follow the steps in the deployment example for guidelines on when to finish setting up your DNS record.

Reviewer: The order of operations felt a little confusing to me here. If most people are going to need to deploy first, and the main takeaway is that you'll need to create the record and most people will need to do it at a specific time, would it be too drastic to simplify it to something like this? I assume this wouldn't negatively impact people who already have their IP address assigned, though if you decide to go this route it'd be good to double check with a technical reviewer.

"You'll need to create a DNS record with your SCIM bridge domain name after the bridge is deployed. Follow the steps in each respective deployment guide to finish setting up your DNS record at the appropriate time."

Author: I think that makes sense!

```suggestion
You'll need to create a DNS record with your SCIM bridge domain name after the bridge is deployed. Follow the steps in each respective deployment guide to finish setting up your DNS record at the appropriate time.
```

With this method, the SCIM bridge must be able to communicate with one of the DNS providers currently supported. As of April 2023, the supported providers are Google Cloud DNS, CloudFlare DNS, and Azure DNS.

Importantly, each DNS service tends to have its own unique way of configuring credentials to authenticate with them. For example, CloudFlare DNS typically only requires an API key (1 credential), whereas Azure DNS requires five separate configuration parameters to fully authenticate with the service.

During the authentication procedure with Let's Encrypt, the SCIM bridge modifies the DNS records of a given name (for example, `op-scim-bridge.example.com`) with certain records that Let's Encrypt is expecting to confirm DNS ownership. Once DNS ownership is confirmed, Let's Encrypt issues a certificate as usual, and the SCIM bridge removes those temporary DNS records.

> To use this method, you'll provide secrets for the DNS service you choose during setup process. You can find an example configuration file at the root of this repository: [`dns01.example.json`](/dns01.example.json).

Reviewer: ```suggestion
To use this method, you'll provide secrets for the DNS service you choose during the setup process. You can find an example configuration file at the root of this repository: [`dns01.example.json`](/dns01.example.json).
```
